### PR TITLE
RichText: fix arrow navigation around emoji

### LINKED ISSUE #27202: Left arrow doesn't jump over emoji on Safari 14.0.1
When navigating text, if there's an emoji on the left side of the cursor pressing left arrow doesn't move it.

**To reproduce**
1. Go to https://wordpress.org/gutenberg/ using Safari 14.0.1 (should be the version on latest MacOS)
2. Insert an emoji in the middle of any paragraph or text field (I have tried with paragraph, captions and more). If you need one you can use 🍓 (have tried with others, too).
3. Use your mouse to place the cursor after the emoji.
4. Press left arrow until you pass the emoji.

**Expected behavior**
The cursor to move past the emoji.

**Screencast**

emoji-left

**Editor version (please complete the following information):**
- WordPress version: 5.6-beta4 (https://wordpress.org/gutenberg/)
- Does the website has Gutenberg plugin installed, or is it using the block editor that comes by default? gutenberg plugin
- If the Gutenberg plugin is installed, which version is it? Not sure

**Desktop (please complete the following information):**
 - OS: Mac OS
 - Browser: Safari
 - Version: 14.0.1

**Additional context**
- Doesn't happen on Firefox at all, on Chrome it sometimes takes a couple of presses of left arrow before it jumps over the emoji, but haven't debugged deeper.

Thanks!

### PR DESCRIPTION
**Pull request by ellatrix**
## Description

Fixes #27202.

The problem is that we're handling (simple) arrow key navigation ourselves. We should only handle it when navigating around format boundaries and otherwise let the browser do its thing.

## How has this been tested?

See #27202.
## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] I've tested my changes with keyboard and screen readers. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

File: packages/e2e-tests/specs/editor/various/writing-flow.test.js [hunk 1]
```diff
@@ -216,6 +216,7 @@ describe( 'Writing Flow', () => {
 		await page.keyboard.up( 'Shift' );
 		await pressKeyWithModifier( 'primary', 'i' );
 		await page.keyboard.press( 'ArrowLeft' );
+		await page.keyboard.press( 'ArrowLeft' );
```
Comment (ellatrix) [inline packages/e2e-tests/specs/editor/various/writing-flow.test.js:219]: This makes sense. First you need to navigate out of italics then out of bold. The old behaviour doesn't make sense.

**Comment by ellatrix** [comment]
This tests well and all rich text e2e tests are passing.

**Comment by nb** [comment]
Thanks, @ellatrix 🍓

**Comment by ellatrix** [comment]
@mkaz Just curious, what is https://github.com/WordPress/gutenberg/pull/30206#ref-commit-749a33b?

**Comment by mkaz** [comment]
@ellatrix I don't know what happened or why the two PRs are still referenced together. I messed up the [rebase on my PR](https://github.com/WordPress/gutenberg/pull/30100), but fixed it up. All the other commits got cleaned up except this one got intermingled, but shows no diff, with this merged I'll remove it from the other PR.

**Comment by ellatrix** [comment]
Ah, ok :) Was just wonder how it got to: "This commit does not belong to any branch on this repository."

**Pull request merged**